### PR TITLE
doc: update two scheduling algorithm

### PR DESCRIPTION
**Pull request by ssqre**
Add description for the IORR (IO sensitive Round Robin) scheduler
and the priority based scheduler.

Signed-off-by: Jie Deng <jie.deng@intel.com>

### CODE REVIEW

**Review by gvancuts — CHANGES_REQUESTED** [review]
* @dbkinder , I find the introduction a bit misleading, is it just me? (this was like this before this PR but I'm thinking we may want to update it)
* The picture in the "Scheduling Framework" section should be updated to show the two other scheduler available.
* There is a **TODO** in the "BVT scheduler" section, is it still applicable?
```
TODO: BVT scheduler will be built on top of prioritized scheduling mechanism, i.e. higher priority threads get scheduled first, and same priority tasks are scheduled per BVT.

    Virtual time: The thread with the earliest effective virtual time (EVT) is dispatched first.
    Warp: a latency-sensitive thread is allowed to warp back in virtual time to make it appear earlier. It borrows virtual time from its future CPU allocation and thus does not disrupt long-term CPU sharing
    MCU: minimum charging unit, the scheduler account for running time in units of MCU.
    Weighted fair sharing: each runnable thread receives a share of the processor in proportion to its weight over a scheduling window of some number of MCU.
    C: context switch allowance. Real time by which the current thread is allowed to advance beyond another runnable thread with equal claim on the CPU. C is similar to the quantum in conventional timesharing.
```
*

**Comment by dbkinder** [comment]
> * There is a **TODO** in the "BVT scheduler" section, is it still applicable?
> 
> ```
> TODO: BVT scheduler will be built on top of prioritized scheduling mechanism, i.e. higher priority threads get scheduled first, and same priority tasks are scheduled per BVT.
> ```

In PR #6591, a comment was added about this saying, 
 
> I will send a another patch to update this doc.
>
> This description of SCHED_BVT is OK. It may have its own prioritized mechanism which is differnt with this priority based scheduler.

I think this means we should remove the TODO content in this PR then.

File: doc/tutorials/cpu_sharing.rst [hunk 1]
```diff
@@ -99,6 +100,21 @@ and BVT (Borrowed Virtual Time) scheduler.
...
+- **Priority based scheduler**:
+
+  The priority based scheduler can support vCPU scheduling based on their
+  pre-configured priorities. A vCPU can be running only if there is no
+  higher priority vCPU running on the same pCPU. For example, in some cases,
+  we have two VM, one VM can be configured to use **PRIO_LOW** and the
+  other one to use **PRIO_HIGH**, then the vCPU of the **PRIO_LOW** VM can
+  only be running when the vCPU of the "PRIO_HIGH" VM voluntarily relinquish
+  the usage of the pCPU.
+
```
Comment (dbkinder) [inline doc/tutorials/cpu_sharing.rst:117]: @ssqre Because of your comment in #6591, we should delete the TODO content on lines 124-127 below, right?

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by dbkinder** [comment]
We should not hold this doc change until after the 2.6 release

File: doc/tutorials/cpu_sharing.rst (hunk 1 above)
Comment (ssqre) [reply doc/tutorials/cpu_sharing.rst:117]: I do agree to remove the TODO content in BVT doc. We don't have any plan to support any priority mechanism for BVT at this moment. It's better to update the BVT doc directly once we update any extension for BVT.

**Comment by dbkinder** [comment]
With the 2.6 release tagged, we can continue with this PR on master

**Comment by dbkinder** [comment]
@NanlinXie @ssqre Can we complete this PR soon?

**Comment by yfliuuu** [comment]
@dbkinder Since @ssqre leaves Intel, I'll take over his doc PR. I'll re-submit a new PR that contains all his changes, and deletes the TODO from 124-127 in cpu_sharing.rst. Any comments?

**Comment by yfliuuu** [comment]
Close this and reopen in https://github.com/projectacrn/acrn-hypervisor/pull/6900

**Pull request closed without merging**